### PR TITLE
feat(ecs): support private registry credentials

### PR DESCRIPTION
**Pull request by clareliguori**
Addresses spinnaker/spinnaker#3339

Related to https://github.com/spinnaker/clouddriver/pull/2998

### CODE REVIEW

**Comment by clareliguori** [comment]
screenshot from 2018-09-27 15-53-48

**Comment by clareliguori** [comment]
Rerun build

**Comment by jrsquared** [comment]
something odd trying to download yarn during tests it seems...
Also, please remove package-lock.json. We use `yarn` for deck.

File: app/scripts/modules/ecs/src/serverGroup/configure/#serverGroupConfiguration.service.ts# [hunk 1]
```diff
@@ -0,0 +1,497 @@
+import { module, IPromise, IQService } from 'angular';
```
Comment (christopherthielen) [inline app/scripts/modules/ecs/src/serverGroup/configure/#serverGroupConfiguration.service.ts#]: What is this file?  It looks to be a dupe of `app/scripts/modules/ecs/src/serverGroup/configure/serverGroupConfiguration.service.ts` but with pound signs in the filename?

Comment (iiro) [reply app/scripts/modules/ecs/src/serverGroup/configure/#serverGroupConfiguration.service.ts#]: Looks like an Emacs buffer auto-save... :)

Comment (clareliguori) [reply app/scripts/modules/ecs/src/serverGroup/configure/#serverGroupConfiguration.service.ts#]: Whoops you are right, removed

Comment (christopherthielen) [reply app/scripts/modules/ecs/src/serverGroup/configure/#serverGroupConfiguration.service.ts#]: I didn't know.  I use vim, like a normal person.  ;)

**Pull request merged**